Lib/idlelib/query.py

```diff
@@ -302,10 +303,61 @@ def entry_ok(self):
         path = self.path_ok()
         return None if name is None or path is None else (name, path)
 
+class CustomRun(Query):
+    "Get arguments for custom run module."
+    # Used in runscript.run_custom_event
+
+    def __init__(self, parent, title, *, cli_args='',
+                 _htest=False, _utest=False):
+        """Get command line args for custom run environment.
+
+        User enters the command line arugments for running the file.
+        """
+        message = 'Command Line Arguments:'
+        super().__init__(
+                parent, title, message, text0=cli_args,
+                _htest=_htest, _utest=_utest)
+
+    def create_widgets(self):
+        super().create_widgets(ok_text='Run')
+        frame = self.frame
+        self.restartvar = BooleanVar(self, value=True)
+        restart = Checkbutton(frame, variable=self.restartvar, onvalue=True,
+                              offvalue=False, text='Restart shell')
+        self.args_error = Label(frame, text=' ', foreground='red',
+                                font=self.error_font)
+
+        restart.grid(column=0, row=4, columnspan=3, padx=5, sticky='w')
+        self.args_error.grid(column=0, row=12, columnspan=3, padx=5,
+                             sticky='we')
+
+    def cli_args_ok(self):
+        "Validity check and parsing for command line arguments."
+        cli_args = self.entry.get().strip()
+        if not cli_args:
+            self.showerror('no arguments specified.')
+            return None
```

Comment:
Not sure about this edit, but it makes everything a little nicer right now.  The reason I'm not sure about it is due to the behavior of the current `Run Module`.  Should `Run Module` use the last settings from `Custom Run` or should it work exactly as it works now?